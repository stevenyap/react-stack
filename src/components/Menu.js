/**
 * @flow
 * Responsibility: Display the menu
 */
import React from 'react'
import { Link } from 'react-router-dom'

export const Menu = () => {
  return (
    <ul>
      <li><Link to="/">Home</Link></li>
    </ul>
  )
}

export default Menu
