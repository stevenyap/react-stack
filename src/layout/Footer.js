/**
 * @flow
 * Responsibility: Render the footer
 */
import React from 'react'
import { Grid, Row, Col } from 'react-bootstrap/lib'

import Colors from 'theme/Colors'

export const Footer = () => {
  return (
    <div style={styles.footer}>
      <Grid>
        <Row>
          <Col>
            Â© React Stack. All rights reserved.
          </Col>
        </Row>
      </Grid>
    </div>
  )
}

const styles = {
  footer: {
    marginTop: 30,
    padding: 10,
    background: Colors.orange,
    color: Colors.white,
    textAlign: 'center'
  }
}

export default Footer
