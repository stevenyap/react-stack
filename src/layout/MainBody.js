/**
 * @flow
 * Responsibility: Render the main body of the app
 */
import React from 'react'
import { Switch, Route } from 'react-router-dom'
import Grid from 'react-bootstrap/lib/Grid'

import Home from 'pages/Home'

export const MainBody = () => {
  return (
    <Grid style={styles.container}>
      <Switch>
        <Route exact={true} path="/" component={Home} />
      </Switch>
    </Grid>
  )
}

const styles = {
  container: {
    flex: '1'
  }
}

export default MainBody
