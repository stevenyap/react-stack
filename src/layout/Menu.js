/**
 * @flow
 * Responsibility: Display the menu
 */
import type { ContextRouter } from 'react-router-dom'

import React from 'react'
import { connect } from 'react-redux'
import { Nav, Navbar, NavItem } from 'react-bootstrap/lib'
import { Link, withRouter } from 'react-router-dom'

import { doLogout } from 'ducks/admin'

type MenuLinkProps = { to: string, text: string } & ContextRouter
const MenuLink = withRouter((props: MenuLinkProps) => {
  const { to, text, history } = props
  const handleClick = () => history.push(to)
  return <NavItem href="#" onClick={handleClick}>{text}</NavItem>
})

type ComponentState = { admin: Admin }
type Props = ComponentState & { ispatch: Dispatch }
export const Menu = (props: Props) => {
  const { admin, dispatch } = props
  const logout = () => dispatch(doLogout())

  return (
    <Navbar inverse={true} collapseOnSelect={true}>
      <Navbar.Header>
        <Navbar.Brand>
          <Link to="/">React Stack</Link>
        </Navbar.Brand>
        {admin && <Navbar.Toggle />}
      </Navbar.Header>
      {admin &&
        <Navbar.Collapse>
          <Nav pullRight>
            <MenuLink to="/" text="Home" />
            <NavItem href="#" onClick={logout}>Logout</NavItem>
          </Nav>
        </Navbar.Collapse>}
    </Navbar>
  )
}

const mapStateToProps = (state: State): ComponentState => ({
  admin: state.admin.data
})
export default connect(mapStateToProps)(Menu)
