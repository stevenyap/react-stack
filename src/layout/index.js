/**
 * @flow
 * Responsibility: Render the layout of the app
 */
import React from 'react'

import Menu from 'layout/Menu'
import MainBody from 'layout/MainBody'

export const Layout = () => {
  return (
    <div>
      <Menu />
      <MainBody />
    </div>
  )
}

export default Layout
