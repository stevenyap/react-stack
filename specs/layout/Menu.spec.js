import React from 'react'
import { BrowserRouter } from 'react-router-dom'

import { Menu } from 'layout/Menu'

describe('Menu', () => {
  it('renders correctly for public', () => {
    const tree = renderJSON(
      <BrowserRouter>
        <Menu />
      </BrowserRouter>
    )
    expect(tree).toMatchSnapshot()

    const renderedText = toText(tree)
    expect(renderedText).toContain('React Stack')
  })

  it('renders correctly for registered users', () => {
    const admin = { uid: 'UID' }
    const tree = renderJSON(
      <BrowserRouter>
        <Menu admin={admin} />
      </BrowserRouter>
    )
    expect(tree).toMatchSnapshot()

    const renderedText = toText(tree)
    expect(renderedText).toContain('React Stack')
    expect(renderedText).toContain('Logout')
  })
})
