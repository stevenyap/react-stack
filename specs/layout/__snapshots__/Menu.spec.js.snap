// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly for public 1`] = `
<nav
  className="navbar navbar-inverse"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        React Stack
      </a>
    </div>
  </div>
</nav>
`;

exports[`Menu renders correctly for registered users 1`] = `
<nav
  className="navbar navbar-inverse"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        React Stack
      </a>
      <button
        className="navbar-toggle collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="sr-only"
        >
          Toggle navigation
        </span>
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
        <span
          className="icon-bar"
        />
      </button>
    </div>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <ul
        className="nav navbar-nav navbar-right"
        role={null}
      >
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
          >
            Home
          </a>
        </li>
        <li
          className=""
          role="presentation"
          style={undefined}
        >
          <a
            href="#"
            onClick={[Function]}
            role="button"
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
